src/plugins/lua/LuaEvent.cpp

```diff
@@ -43,40 +43,56 @@ LuaEventProcessor::~LuaEventProcessor()
 	m_eventDefinitions.clear();
 }
 
-void LuaEventProcessor::AddEvent(std::string_view name, std::string_view expression, const sol::function& function)
+bool LuaEventProcessor::AddEvent(std::string_view name, std::string_view expression, const sol::function& function)
 {
+	// the number of events will always be fairly small, and this is a manual operation.
+	// If this is deemed too slow, the event names can be memoized in a set of string_views.
+	auto it = std::find_if(m_eventDefinitions.begin(), m_eventDefinitions.end(),
+		[&name](const std::unique_ptr<LuaEvent>& event) { return event->GetName() == name; });
+
+	if (it != m_eventDefinitions.end())
+	{
+		LuaError("Cannot create event %.*s, it is already defined.", name.length(), name.data());
+		return false;
+	}
+
 	m_eventDefinitions.push_back(std::make_unique<LuaEvent>(name, expression, function, this, *m_blech));
+	return true;
 }
 
-void LuaEventProcessor::RemoveEvent(std::string_view name)
+bool LuaEventProcessor::RemoveEvent(std::string_view name)
 {
 	RemoveEvents({ std::string(name) });
 	auto it = std::find_if(m_eventDefinitions.begin(), m_eventDefinitions.end(),
 		[&name](const std::unique_ptr<LuaEvent>& event) { return event->GetName() == name; });
 	if (it != m_eventDefinitions.end())
 	{
 		m_eventDefinitions.erase(it);
+		return true;
```

Comment:
is this list we're iterating through only the ones with matching ids?  If not, how do we know it's the last one?

Reply:
I'm not sure I understand. It's finding a single entry that is guaranteed to have a unique name because of the main change in this PR.

Reply:
ah true, this is now assuming unique keys